public/components/integrations/components/setup_integration.tsx

```diff
@@ -118,10 +119,52 @@ const suggestDataSources = async (type: string): Promise<Array<{ label: string }
   }
 };
 
-const findTemplate = async (integrationTemplateId: string) => {
-  const http = coreRefs.http!;
-  const result = await http.get(`${INTEGRATIONS_BASE}/repository/${integrationTemplateId}`);
-  return result;
+const runQuery = async (
+  query: string,
+  trackProgress: (step: number) => void
+): Promise<Result<object>> => {
+  // Used for polling
+  const sleep = (ms: number) => {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  };
+
+  try {
+    console.log('- Posting query');
+    const http = coreRefs.http!;
+    const queryId = (
+      await http.post(CONSOLE_PROXY, {
+        body: JSON.stringify({ query, lang: 'sql' }),
+        query: {
+          path: '_plugins/_async_query',
+          method: 'POST',
+        },
+      })
+    ).queryId;
+    while (true) {
```

Comment:
Is there better way than while true?

Reply:
Not sure what kind of polling timeout to put on it, I've heard some queries can take up to 10 mins and don't have any reference for when I should just tell the user that it isn't working